auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -403,17 +407,111 @@ public static AuthUI getInstance(FirebaseApp app) {
         return R.style.FirebaseUI;
     }
 
+
+    /**
+     * Configuration for an identity provider.
+     *
+     * In the simplest case, you can supply the provider ID and build the config like this:
+     * {@code new IdpConfig.Builder(AuthUI.GOOGLE_PROVIDER).build()}
+     */
+    public static class IdpConfig implements Parcelable {
```

Comment:
Would it be better if each provider had a subclass of this?  so `new GoogleIdpConfig.Builder().build()`, etc.  This way you have more control over provider differences.  For example doesn't look like you're supporting Twitter scopes so that builder would not have `setScopes()`.  It would also hide the `GOOGLE_PROVIDER` and other Strings from the dev.

This one is up to you, I can see arguments both ways.